### PR TITLE
chore(shadertools): Port all shaders to GLSL 3.00, enforce GLSL 3.00 source

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Background
- As I started making changes to shader generation, I noticed that maintaining the GLSL 1.00 -> GLSL 3.00 transpilation path was taking a lot of effort
- Given that organizing the code around uniform buffers is required for WebGPU compatibility we might as well enforce that all shader source is 3.00, and only support transpilation from 3.00 -> 1.00
- Enforcing that led to a number of code fragments across source and tests being upgraded to 3.00
#### Change List
- As above
- Improve device info GLSL version handling
- Make the existing sloppy shader version handling more strict across the board, fix cases where version was just assumed.
- Update tests.

### CODE REVIEW

File: docs/api-reference/core/device.md [hunk 1]
```diff
@@ -148,19 +148,26 @@ info: DeviceInfo
...
-| `renderer`               | `string` | Renderer (unmasked if possible)   |
-| `version`                | `string` | WebGL version                     |
-| `shadingLanguageVersion` | `string` | shading language version          |
+| Field                    | Type     | Description                           |
+| ------------------------ | -------- | ------------------------------------- |
+| `vendor`                 | `string` | GPU vendor (unmasked if possible)     |
+| `renderer`               | `string` | Renderer (unmasked if possible)       |
+| `version`                | `string` | WebGL version                         |
+| `gpu`                    | `string` | GPU name                              |
+| `gpuBackend?`            | `string` | `'angle' \| 'metal' \| 'unknown'`     |
```
[the remark this answers is not in the document]
Comment (donmccurdy) [reply docs/api-reference/core/device.md:159]: On my (Apple M2) device, here are my unmasked renderers in Chrome:

- with ANGLE's OpenGL backend: `ANGLE (Apple, Apple M2 Pro, OpenGL 4.1)`
- with ANGLE's Metal backend: `ANGLE (Apple, ANGLE Metal Renderer: Apple M2 Pro, Unspecified Version)`

And in Safari:

- `Apple GPU`

My understanding is that modern Chrome and Safari always use ANGLE today, but that Chrome chooses between ANGLE's OpenGL and Metal backends "somehow", and Safari always uses ANGLE's Metal backend.

So I'm not sure `'angle' | 'metal' | 'unknown'` is the right set of choices here, maybe `'angle-metal' | 'angle-opengl' | 'unknown'` ?

File: modules/core/src/adapter/device.ts [hunk 2]
```diff
@@ -54,46 +54,30 @@ export type DeviceProps = {
...
   vendor: string;
   renderer: string;
+  /** version of driver */
   version: string;
+  /** type of GPU */
   gpu: 'nvidia' | 'amd' | 'intel' | 'apple' | 'software' | 'unknown';
-  shadingLanguages: ShadingLanguage[];
-  shadingLanguageVersions: Record<string, string>;
+  /** GPU driver backend. Can sometimes be sniffed */
+  gpuBackend?: 'angle' | 'metal' | 'unknown';
```
Comment (donmccurdy) [inline modules/core/src/adapter/device.ts:73]: See previous comment.

**Review by donmccurdy — APPROVED** [review]
Left some thoughts about the `gpuBackend` property, but this all looks great!

File: docs/api-reference/core/device.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibgreen) [reply docs/api-reference/core/device.md:159]: Yes.. `metal` vs `opengl` makes sense to me! Yes I suspected that we'd need to polish these and the detection tests. Feel free to modify if you happen to touch more GPU tests.

**Pull request merged**